File: app/src/main/java/net/mmho/photomap2/ThumbnailCache.java
package net.mmho.photomap2;

import android.graphics.Bitmap;
import android.support.v4.util.LruCache;

public class ThumbnailCache extends LruCache<Long,Bitmap> {

    static private ThumbnailCache self;
    private static final int maxMemory = (int)(Runtime.getRuntime().maxMemory()/1024);
    private static final int cacheSize = maxMemory/8;

    private ThumbnailCache(int maxSize) {
        super(maxSize);
    }

    static public ThumbnailCache getInstance(){
        if(self==null){
            self = new ThumbnailCache(cacheSize);
        }
        return self;
    }

    @Override
    protected int sizeOf(Long key, Bitmap bitmap) {
        return bitmap.getRowBytes() * bitmap.getHeight() / 1024;
    }
}

File: app/src/main/kotlin/net/mmho/photomap2/ThumbnailCache.kt
package net.mmho.photomap2

import android.graphics.Bitmap
import android.support.v4.util.LruCache

class ThumbnailCache private constructor(maxSize: Int) : LruCache<Long, Bitmap>(maxSize) {

    override fun sizeOf(key: Long?, bitmap: Bitmap): Int {
        return bitmap.rowBytes * bitmap.height / 1024
    }

    private object Self { val INSTANCE = ThumbnailCache(cacheSize) }

    companion object {
        val instance:ThumbnailCache by lazy { Self.INSTANCE }

        private val maxMemory = (Runtime.getRuntime().maxMemory() / 1024).toInt()
        private val cacheSize = maxMemory / 8

    }
}
